hack/verify-docker-build.sh

```diff
@@ -23,7 +23,4 @@ source "$(dirname "$0")/utils.sh"
 # check if manager docker image builds
 cd_root_path
 
-export GO111MODULE=on
-go mod download
-go build -o manager ./cmd/manager
```

Comment:
These are no longer needed due to the fact that the [Dockerfile builds the binary](https://github.com/kubernetes-sigs/cluster-api-provider-docker/blob/0823568185a4d983ff4c36ececea7cc2ac3acd3e/Dockerfile#L34) and [copies it via multi-stage builds](https://github.com/kubernetes-sigs/cluster-api-provider-docker/blob/0823568185a4d983ff4c36ececea7cc2ac3acd3e/Dockerfile#L54)